api/proto/certs/certs.proto

```diff
@@ -32,23 +21,15 @@ enum certHashAlgorithm {
     HASH_NONE = 0;
     HASH_SHA256_16bytes = 1;  // hash with sha256, the 1st 16 bytes of result in 'certHash'
 }
+
 message ZCert {
    certHashAlgorithm hashAlgo = 1; //hash method used to arrive at certHash
-   bytes certHash = 2;             //Hash of the cert, computed using hashAlgo
-   ZCertType type = 3;             //what kind of certificate(to identify the target use case)
-   bytes cert = 4;                 //X509 cert in .PEM format
-   ZCertProperties properties = 5; //properties of this certificate
+   ZCertType type = 2;             //what kind of certificate(to identify the target use case)
+   bytes cert = 3;                 //X509 cert in .PEM format
 }
 
 enum ZCertType {
-   CERT_TYPE_DEVICE_ONBOARDING = 0;         //set for certificate used by edge node for identifying the device
-   CERT_TYPE_DEVICE_RESTRICTED_SIGNING = 1; //set for certificate used by edge node for attestation
-   CERT_TYPE_DEVICE_ENDORSEMENT_RSA = 2;    //set for endorsement key certificate with RSASSA signing algorithm
-   CERT_TYPE_DEVICE_ECDH_EXCHANGE = 3;      //set for certificate used by edge node to share symmetric key using ECDH
-   CERT_TYPE_CONTROLLER_SIGNING = 4;        //set for certificate used by controller to sign payload envelopes
-   CERT_TYPE_CONTROLLER_ECDH_EXCHANGE = 5;  //set for certificate used by controller to share any symmetric key using ECDH
-}
-
-message ZCertProperties {
-   bool isMutable = 1; //set to false for immutable certificates
+   CERT_TYPE_CONTROLLER_SIGNING = 0;        //set for the leaf certificate used by controller to sign payload envelopes
+   CERT_TYPE_CONTROLLER_INTERMEDIATE = 1;   //set for intermediate certs used to sign the leaf certificate
```

Comment:
Comment is incorrect AFAIK, since the same intermediates are likely to be used to validate the ECDH_EXCHANGE cert.

Reply:
Edited it, to reflect that intermediate certificates can be used to validate any certificate.